lib/babel.js

```diff
@@ -74,8 +77,33 @@ if (inputSourceMap) {
 	options.inputSourceMap = JSON.parse(inputSourceMap.map);
 }
 
+// load up custom compilers
+process.argv.slice(3).forEach(function (arg) {
+	var match = /^--compilers=.+?:(.+)$/i.exec(arg);
```

Comment:
> Mocha's watcher does weird things when it reloads the tests

By nature of the way we fork new processes for each test, that should not be an issue for us.

>  I'm not convinced watching specific extensions is even necessary.

There is [discussion](https://github.com/sindresorhus/ava/issues/115) of including an option to have the watcher only run tests affected by the change. Though perhaps a better solution would be to just monitor the keys of `Module._extensions`, or allow a list of extensions to be passed to the watcher flag `--watch=jsx,coffee,...`